helm/linkerd2-app/tests/abs/test_basic_cluster.py

```diff
@@ -0,0 +1,174 @@
+linkerd_namespace = "linkerd"
+linkerd_app_name = "linkerd2-app"
+cni_namespace = "default"
+cni_app_name = "linkerd2-cni-app"
+test_app_namespace = "helloworld"
+
+
+def get_linkerd_cli(version) -> None:
+    url = f"https://github.com/linkerd/linkerd2/releases/download/{version}/linkerd2-cli-{version}-linux-amd64"
+    logger.info(f"Downloading linkerd-cli from url '{url}'")
+    local_filename = "linkerd-cli"
+    if os.path.isfile(local_filename) and os.access(local_filename, os.X_OK):
+        return
+    with requests.get(url, stream=True) as r:
+        with open(local_filename, 'wb') as f:
+            shutil.copyfileobj(r.raw, f)
+
+    os.chmod(local_filename, 0o755)
+
+
+def exec_linkerd_cli(kube_config_path, namespace_cni, namespace, app_namespace) -> Any:
+    result = subprocess.run([
+        "./linkerd-cli",
+        "check",
+        "--kubeconfig", kube_config_path,
+        "--cni-namespace", namespace_cni,
+        "--linkerd-namespace", namespace,
+        "--proxy",
+        "--namespace", app_namespace,
+        "--output", "json",
+    ],
+        encoding="utf-8",
+        stdout=subprocess.PIPE,
+        stderr=None,
+    ).stdout
+
+    return json.loads(result)
+
+
+def load_yaml_from_path(filepath) -> Any:
+    with open(filepath, 'r', encoding='utf-8') as values_file:
+        values = values_file.read()
+
+    yaml_config = yaml.safe_load(values)
+    return yaml_config
+
+
+def wait_for_all_linkerd_deployments_to_run(kube_client: HTTPClient, namespace: str, timeout_sec: int) -> None:
+    wait_for_deployments_to_run(
+        kube_client,
+        [
+            "linkerd-controller",
+            "linkerd-destination",
+            "linkerd-identity",
+            "linkerd-proxy-injector",
+            "linkerd-sp-validator",
+        ],
+        namespace,
+        timeout_sec,
+    )
+
+
+@pytest.fixture(scope="module")
+def linkerd_cni_app_cr(app_factory: AppFactoryFunc) -> ConfiguredApp:
+    res = app_factory(cni_app_name,
+                      cni_app_version,
+                      "giantswarm-test",
+                      "https://giantswarm.github.io/giantswarm-test-catalog/",
+                      timeout_sec=timeout,
+                      namespace=cni_namespace,
+                      namespace_config_annotations={"linkerd.io/inject": "disabled"},
+                      namespace_config_labels={
+                          "linkerd.io/cni-resource": "true",
+                          "config.linkerd.io/admission-webhooks": "disabled"
+                      }
+                      )
+    return res
+
+
+@pytest.fixture(scope="module")
+def linkerd_app_cr(kube_cluster: Cluster, app_factory: AppFactoryFunc, chart_version: str,
+                   linkerd_cni_app_cr: ConfiguredApp) -> ConfiguredApp:
+    ensure_namespace_exists(kube_cluster.kube_client, linkerd_namespace)
+    res = app_factory(linkerd_app_name,
+                      chart_version,
+                      "chartmuseum-test-time",
```

Comment:
It this right?

Reply:
Yes, as the fixture doesn't detect nor manage pre-created appCatalogs, so it can't manage the one created earlier `apptestctl`. So I'm registering the same catalog here, just with different name to avoid name conflict.

Reply:
ok, maybe you can add a comment for future reference (my cheese brain)